File: JMMinistry.Common/Dtos/User/PartialUserInfoDto.cs
using JMMinistry.Common.Dtos.User.Enums;

namespace JMMinistry.Common.Dtos.User
{
    public class PartialUserInfoDto
    {
        public string Document { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Phone { get; set; } = string.Empty;
        public MinistryStatus? MinistryStatus { get; set; }
        public Gender Gender { get; set; }
        public string? Photo { get; set; }
        public int? CellId { get; set; }
    }
}

File: the_vine_hub_app/lib/shared/domain/models/partial_user_info_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'partial_user_info_dto.freezed.dart';
part 'partial_user_info_dto.g.dart';

@freezed
abstract class PartialUserInfoDto with _$PartialUserInfoDto {
  const PartialUserInfoDto._();

  const factory PartialUserInfoDto({
    String? document,
    String? name,
    String? lastName,
    String? phone,
    int? gender,
    int? maritalStatus,
    String? photo,
    int? cellId,
  }) = _PartialUserInfoDto;

  factory PartialUserInfoDto.fromJson(Map<String, dynamic> json) => _$PartialUserInfoDtoFromJson(json);
}
